desktop/core/src/desktop/js/reactComponents/FileUploadQueue/FileUploadQueue.tsx

```
@@ -66,28 +86,98 @@ const FileUploadQueue = (): JSX.Element => {
     onComplete
   });
 
+  // If there are already files in the queue (e.g., restored state), ensure the queue is visible
+  useEffect(() => {
+    if (uploadQueue.length > 0) {
+      setIsVisible(true);
+    }
+  }, [uploadQueue.length]);
+
   useHuePubSub<FileUploadEvent>({
     topic: FILE_UPLOAD_START_EVENT,
-    callback: (data?: FileUploadEvent) => {
-      if (data?.files) {
-        setIsVisible(true);
-        addFiles(data.files);
+    callback: async (data?: FileUploadEvent) => {
+      const newFiles = data?.files ?? [];
+      if (newFiles.length > 0) {
```

Don't we need to handle the case where newFiles.length === 0? When does that happen and how should the system then react? Now it looks like it fails silently... if the case is handled earlier, then I would assume the we should not even trigger the FileUploadEvent

---

Makes Sense. i added a conditional logic to take care of this.

---

updated.